app/main.py

```diff
@@ -1,6 +1,18 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+
+    cache_storage = {}
+
+    def inner(*args) -> Any:
+        if args in cache_storage:
+            print("Getting from cache")
+            return cache_storage[args]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            cache_storage[args] = result
+            return result
```

Comment:
This else look useless 
```suggestion
        print("Calculating new result")
        result = func(*args)
        cache_storage[args] = result
        return result
```